src/legacy/core_plugins/embeddable_api/public/actions/action.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { EuiContextMenuItemIcon } from '@elastic/eui';
+import { trackUiMetric } from '../../../ui_metric/public';
+
+import { IEmbeddable } from '../embeddables';
+
+export interface ExecuteActionContext<
```

Comment:
Hm, so... I realized I can get rid of this interface entirely, for in phase 1 you will always have the trigger context (since this is run when you are actually about to execute the action)...

Only in phase 2 you might not have the full trigger context, because an end user could be setting up an action, and in that case you still want to know if an action is compatible with the embeddable, but you don't have the trigger context ahead of time.

hmmm....

Reply:
I'm going to go ahead and consolidate the two interfaces for phase 1.  For phase 2, we can change the interface to make `triggerContext` optionally available, or maybe use some other method of determining compatibility with an embeddable.